base_layer/core/src/proof_of_work/proof_of_work.rs

```diff
@@ -98,10 +99,10 @@ impl ProofOfWork {
     ///
     /// If there are any problems with calculating a difficulty (e.g. an invalid header), then the function returns a
     /// difficulty of one.
-    pub fn achieved_difficulty(header: &BlockHeader) -> Difficulty {
+    pub fn achieved_difficulty(header: &BlockHeader) -> Result<Difficulty, PowError> {
         match header.pow.pow_algo {
-            PowAlgorithm::Monero => monero_difficulty(header),
-            PowAlgorithm::Blake => blake_difficulty(header),
+            PowAlgorithm::Monero => Ok(monero_difficulty(header)?),
+            PowAlgorithm::Blake => Ok(blake_difficulty(header)),
```

Comment:
Should we not get this error out as well?

Reply:
this error?

Reply:
If anything goes wrong inside of blake, it suppresses the error and returns 0

Reply:
Didn't see any error there

Reply:
@SWvheerden Think you may be confusing this with the previous implementation of `monero_difficulty` which did suppress the error. `blake_difficulty` doesn't return an error at any point AFAICT.